### PR TITLE
pythagorean-triplets: implement canonical-data.json

### LINKED ISSUE #579: pythagorean-triplet: Implement canonical-data.json
We want to have a standard set of test inputs and outputs for each exercise to
make it easier to port them to new languages, as well as to help keep the
exercises in sync.

The Pythagorean Triplet exercise can be found in the [./exercises/pythagorean-triplet/][exercise-dir] directory.

The step-by-step instructions for how to compile a canonical-data.json file is
described in [this section of the contributing guide][contributing].

See http://exercism.io/contribute/canonical-data/pythagorean-triplet for the up-to-date list of
language tracks that have an implementation of the Pythagorean Triplet exercise.

The [tracking issue][tracking-issue] has more context.

[contributing]: https://github.com/exercism/docs/blob/master/you-can-help/improve-exercise-metadata.md#extracting-canonical-test-data
[tracking-issue]: https://github.com/exercism/x-common/issues/552
[exercise-dir]: https://github.com/exercism/x-common/tree/master/exercises/pythagorean-triplet

### LINKED ISSUE #1211: pythagorean-triplet: improve readme and specification
@yawpitch submitted this over in https://github.com/exercism/exercism.io/issues/3743

---- 

I'm working on the Python track specifically, but I notice that across many of the language tracks the tests and examples do not reflect the description of the problem:

> Pythagorean Triplet
> A Pythagorean triplet is a set of three natural numbers, {a, b, c}, for which,
> `a**2 + b**2 = c**2`
> For example,
> `3**2 + 4**2 = 9 + 16 = 25 = 5**2`.
> There exists exactly one Pythagorean triplet for which a + b + c = 1000.
> Find the product `a * b * c`.

In the Python, Rust, and Haskell tracks, for instance, there are no tests whatsoever related to the "find the product `a*b*c`" statement, and all three require different constructs than what's implied by the problem description and title. The Javascript track, however, does require that the work produced can calculate the product.

### PR DESCRIPTION
**Pull request by cmccandless**
Resolves #579

This data set was generated using [this one-off script](https://gist.github.com/cmccandless/94290147674158cc79727f250fcc5613), for those who are curious about where the cases came from, with a runtime of ~21.5 seconds on my machine (primarily due to the test case for a large `n`).

I welcome additional test suggestions or suggestions for modification/removal of currently present cases or case format.

### CODE REVIEW

File: exercises/pythagorean-triplet/canonical-data.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                [200, 375, 425]
+            ]
+        },
+        {
+            "description": "no matching triplets for 1001",
+            "property": "tripletsWithSum",
+            "input": {
+                "n": 1001
+            },
+            "expected": []
```
[the remark this answers is not in the document]
Comment (rpottsoh) [reply exercises/pythagorean-triplet/canonical-data.json]: You might be right, I am just thinking of this [comment](https://github.com/exercism/problem-specifications/issues/1312#issuecomment-417614484) made by @coriolinus.

[the remark this answers is not in the document]
Comment (cmccandless) [reply exercises/pythagorean-triplet/canonical-data.json]: If we were *searching*, I would agree with that comment.

I think of this more as a *filtering* operation: of existing Pythagorean triplets, keep the ones that sum to `n`.

[the remark this answers is not in the document]
Comment (petertseng) [reply exercises/pythagorean-triplet/canonical-data.json]: I expect an empty list. This seems like the proper way to indicate there are no triplets. I reason that there should not be a language for which an empty list does not make sense.

**Comment by petertseng** [comment]
EBWODP:

> * > 1 triplets

That didn't go well. That should be `> 1` triplets.

**Comment by sdublish** [comment]
I'm a bit unclear on how this resolves #1211 as it this does not modify description.md, but other than that, it looks fine to me.

**Comment by cmccandless** [comment]
As there are 3 approvals already, I will go ahead and squash-merge this. Thank you everyone for your input.

@petertseng, note that I have indeed sorted the expected output.

@sdublish You are right, I have removed the link from the PR description so #1211 will remain open.

**Pull request merged**